downstairs/src/region.rs

```diff
@@ -24,13 +23,66 @@ pub struct Extent {
 #[derive(Debug)]
 pub struct Inner {
     file: File,
-    meta: ExtentMeta,
+    metadb: Connection,
+}
+
+impl Inner {
+    fn flush_number(&self) -> Result<u64> {
+        let mut stmt = self
+            .metadb
+            .prepare("SELECT value FROM metadata where name='flush_number'")?;
+        let flush_number_iter = stmt.query_map([], |row| row.get(0))?;
+
+        let mut flush_number_values: Vec<u64> = vec![];
+        for flush_number_value in flush_number_iter {
+            flush_number_values.push(flush_number_value?);
+        }
+
+        assert!(flush_number_values.len() == 1);
+
+        Ok(flush_number_values[0])
+    }
+
+    fn set_new_flush_number(&self, new_flush: u64) -> Result<()> {
+        /*
+         * When we write out the new flush number, the dirty bit in the file
+         * should be set back to false.
+         */
+        let mut stmt = self
+            .metadb
+            .prepare(
+                "UPDATE metadata SET value=?1 WHERE name='flush_number'; UPDATE metadata SET dirty=0",
+            )?;
+
+        let _rows_affected = stmt.execute(params![new_flush])?;
+
+        Ok(())
+    }
+
+    fn dirty(&self) -> Result<bool> {
+        let mut stmt = self
+            .metadb
+            .prepare("SELECT value FROM metadata where name='dirty'")?;
+        let dirty_iter = stmt.query_map([], |row| row.get(0))?;
+
+        let mut dirty_values: Vec<bool> = vec![];
+        for dirty_value in dirty_iter {
+            dirty_values.push(dirty_value?);
+        }
+
+        assert!(dirty_values.len() == 1);
+
+        Ok(dirty_values[0])
+    }
+
+    fn set_dirty(&self) -> Result<()> {
+        let _rows_affected = self
+            .metadb
+            .execute("UPDATE metadata SET value=1 WHERE name='dirty'", [])?;
+        Ok(())
+    }
 }
 
-/*
```

Comment:
Is this comment no longer true?  Or just too obvious to leave here?  Just wondering why it was removed.

Reply:
I think it's no longer true, we're not reading serialized bytes anymore and don't need to version the struct or something like that. If ExtentMeta changes now we'll need something like a schema upgrade. This is especially true if we change the type on the RHS, we would need a new table for that.